02-peer-protocol.md

```diff
@@ -553,11 +553,24 @@ the other node then replies similarly, using a fee it thinks is fair.  This
 exchange continues until both agree on the same fee or when one side fails
 the channel.
 
+In the modern method, the funder sends its permissable fee range, and the
```

Comment:
Don't we need a feature bit for this? As otherwise, an upgraded funder can send these new fields, for it only to be ignored by the non-funder (or the other way around). IMO it's also desirable to be able to _force_ this behavior in the future by having nodes advertise it as a required feature bit. 

Independent of that, it may be desirable for nodes to seek out other nodes that also understand this new behavior as it may allow them to pay lower co-op closing fees on average.

Reply:
Related to the comment below, and also to this feature bit idea: what if we stopped including the signature (for the new co-op close style), until one side fully "accepts" a proposal? As is, you can't actually continue using the channel once a co-op close attempt is initiated, since if it fails, or both sides D/C (or w/e) one side has a valid signature to close out the channel, so they can use that at anytime. On top of that, as is, a sort of signing oracle is exposed, since AFAIK implementations don't really clamp down on the number of iterations during negotiation, so they may just force another node to continue signing new messages which _may_ leak information. 

So if we drop the sigs (say they're all zero) until one side "accepts" (signalled by sending an actual signature), the only valid sigs ever sent across are for the final agreed fee rate. If for w/e reason, negotiation breaks down (neither side sends a sig), then its safe for both sides to use the channel. The other added benefit here is that when the "negotiation" method is used, you don't need to sign so it's more performant CPU-wise.

Reply:
> IMO it's also desirable to be able to force this behavior in the future by having nodes advertise it as a required feature bit.

What does this accomplish? At least on the LDK end we already have (reasonably-deterministically-generated) an internally-stored min/max, exposing that to the counterparty just lets things go quicker.

> what if we stopped including the signature (for the new co-op close style), until one side fully "accepts" a proposal? 

because it breaks backwards compatibility.

Reply:
We've discussed several times why we don't use a feature bit (see previous comments on this PR).
This is quite a simple feature which will simply replace the old negotiation (which we'll completely remove from the spec in X months).

Reply:
> This is quite a simple feature which will simply replace the old negotiation (which we'll completely remove from the spec in X months).

How can it be removed without having a new feature bit? If we have a new bit, we'd simply all switch it over to _required_ which effectively allow us to remove the old logic for these clients.

Reply:
> Nodes in the wild can seek out peers that support this feature so they don't need to implement the "old" negotiation protocol. Nodes that already have the old logic may also want to instead

Do you plan on doing this? If not, then I think no one will and its probably not worth worrying about.

> What's the tangible cost of adding a feature bit here?

Complexity and more stuff to check in a chunk of code that already has a ton of tests.

> How can it be removed without having a new feature bit? If we have a new bit, we'd simply all switch it over to required which effectively allow us to remove the old logic for these clients.

You haven't explained why this is different with or without a feature bit.